tests/integration/common/transaction_utils.rs

```diff
@@ -614,6 +647,65 @@ pub fn to_internal_invoke_v3_tx(tx: &InvokeTransactionV3) -> InternalTransaction
     };
 }
 
+/// Convert a InvokeTransactionV3 to a SNOS InternalTransaction
+pub fn to_internal_deploy_v3_tx(tx: &DeployAccountTransactionV3) -> InternalTransaction {
+    let signature = Some(tx.signature.0.iter().map(|x| to_felt252(x)).collect());
+    let entry_point_selector = Some(Felt252::ZERO);
+    let calldata: Vec<_> = tx.constructor_calldata.0.iter().map(|x| to_felt252(x.into())).collect();
+
+    let nonce = felt_api2vm(tx.nonce.0);
+    let tip = felt_api2vm(tx.tip.0.into());
+
+    let nonce_data_availability_mode = Felt252::from(tx.nonce_data_availability_mode as u64);
+    let fee_data_availability_mode = Felt252::from(tx.fee_data_availability_mode as u64);
+    let resource_bounds = &tx.resource_bounds;
+
+    let paymaster_data: Vec<Felt252> = tx.paymaster_data.0.iter().map(|x| to_felt252(x.into())).collect();
+
+    let sender_address = calculate_contract_address(
+        tx.contract_address_salt,
+        tx.class_hash,
+        &tx.constructor_calldata,
+        ContractAddress::from(0_u8),
```

Comment:
That's the deployer address and it shouldn't be zero in all cases. This field is provided in the tx data, please support it and parametrize the test so that we test the case where the deployer is 0 or if the deployer is the sender.

Reply:
Mmh that's a good point, it's technically not a part of the tx but a part of the contract address... I'm a bit surprised by this, let me check the Cairo code.

Reply:
Okay, not too sure why but this seems to be a feature used only in the `deploy` syscall and not in deploy txs. Interesting, needs to be investigated but we can keep it as it is for the tx tests here.